key.core/src/main/java/de/uka/ilkd/key/proof/io/ProofSaver.java

```diff
@@ -1,20 +1,24 @@
 package de.uka.ilkd.key.proof.io;
 
+import de.uka.ilkd.key.proof.Proof;
+import de.uka.ilkd.key.proof.init.ProblemInitializer;
+import de.uka.ilkd.key.proof.io.event.ProofSaverEvent;
+import de.uka.ilkd.key.proof.io.event.ProofSaverListener;
+import de.uka.ilkd.key.util.KeYConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.util.LinkedList;
 import java.util.List;
 
-import de.uka.ilkd.key.proof.Proof;
-import de.uka.ilkd.key.proof.io.event.ProofSaverEvent;
-import de.uka.ilkd.key.proof.io.event.ProofSaverListener;
-import de.uka.ilkd.key.util.KeYConstants;
-
 /**
  * Saves a proof and provides useful methods for pretty printing terms or programs.
  */
 public class ProofSaver extends OutputStreamProofSaver {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProblemInitializer.class);
```

Comment:
What's our policy towards Loggers? Should that not be `OutputStreamProofSaver.class`?

Reply:
I don't know whether we have one but I think this should definitely be the current class and not the parent class (this one here is wrong, don't know why the cli option missed it).

It's very confusing to get logger messages from `BaseClass` and not find any log statement declared in this class.

Reply:
Perhaps we should make that the rule then. (-> KaKeY)

Reply:
Exception: `SyntaxErrorReporter`, although this is a very odd place to log and I think it should go away.

Reply:
I think this is should be added to the logging doc page, but I'm currently waiting on some changes by @wadoon to land before writing that one.